File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/data/KDataBase.java
package org.kevoree.modeling.api.data;

import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.KEvent;
import org.kevoree.modeling.api.ModelListener;
import org.kevoree.modeling.api.ThrowableCallback;
import org.kevoree.modeling.api.event.DefaultKEvent;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 05/11/2013
 * Time: 11:30
 */

public interface KDataBase {

    public void get(String[] keys, ThrowableCallback<String[]> callback);

    public void put(String[][] payloads, Callback<Throwable> error);

    public void remove(String[] keys, Callback<Throwable> error);

    public void commit(Callback<Throwable> error);

    public void close(Callback<Throwable> error);

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/data/KDataBase.ts
///<reference path="../Callback.ts"/>
///<reference path="../ThrowableCallback.ts"/>

interface KDataBase {

  get(keys: string[], callback: ThrowableCallback<string[]>): void;

  put(payloads: string[][], error: Callback<Throwable>): void;

  remove(keys: string[], error: Callback<Throwable>): void;

  commit(error: Callback<Throwable>): void;

  close(error: Callback<Throwable>): void;

}

